app/scripts/controllers/createClientBuild.js

```diff
@@ -98,21 +105,26 @@ angular.module('openshiftConsole')
                 {
                   name: 'BUILD_CONFIG',
                   value: clientConfig.buildType
+                },
+                {
+                  name: 'PLATFORM',
+                  value: clientConfig.buildPlatform
                 }
               ]
             }
           }
         }
       };
 
-      if(clientConfig.buildType === releaseBuildType.id) {
+      if(clientConfig.clientCredentialsName) {
         buildConfig.spec.strategy.jenkinsPipelineStrategy.env.push({name: 'BUILD_CREDENTIAL_ID', value: $scope.projectName + '-' + clientConfig.clientCredentialsName});
-        if(clientConfig.credentialsAlias) {
-          buildConfig.spec.strategy.jenkinsPipelineStrategy.env.push({name: 'BUILD_CREDENTIAL_ALIAS', value: clientConfig.credentialsAlias});
-        }
       }
 
-      if (clientConfig.authType !== 'public') {
+      if(clientConfig.credentialsAlias) {
+        buildConfig.spec.strategy.jenkinsPipelineStrategy.env.push({name: 'BUILD_CREDENTIAL_ALIAS', value: clientConfig.credentialsAlias});
+      }
+
+      if (clientConfig.gitCredentialsName) {
         buildConfig.spec.source.sourceSecret = {
           name: clientConfig.gitCredentialsName
         };
```

Comment:
Made all these non platform/client type specific, if they are set on the object, do what needs to be done, easier than multiple statements checking platforms and build types.